### DISCUSSION TITLE
KEIL uVision project setup

### DISCUSSION THREAD

**Discussion by DomenGosek** [Broadcom/AFBR-S50-API#28] [General]
Hello Broadcom,

I tried to setup and build a project in KEIL uVision IDE. I ran into some issues and have been wondering if the sensor library builds with ARM compiler v6 or is only buildable with GCC? If you tried and suceeded with any example including sensor library to uVision and build with ARM compiler is it possible to share?

Best regards

**Comment by c-berger** [comment]
Hi, the library is built with GCC but it should be linkable with any other toolchain if it support AEABI compatible library linking.

Please have a look in the API reference manual for more information (see Toolchain Compatibility):
https://broadcom.github.io/AFBR-S50-API/porting_guide.html

Or directly read more about the topic here:
https://developer.arm.com/documentation/dai0487/latest/

Note that we have not tested this. But we know from other customers that it basically is possible as they have been able to successfully link the library.

Best regards!
